protocols/mdns/src/query.rs

```diff
@@ -269,10 +253,13 @@ impl MdnsPeer {
             })
             .collect();
 
-        MdnsPeer {
-            addrs,
-            peer_id: my_peer_id,
-            ttl,
+        match my_peer_id {
```

Comment:
nit: you can simplify this a bit by doing:
```
peer_id.map(|peer_id| MdnsPeer {addrs, peer_id, ttl})
```

Reply:
Great catch. I never knew that.